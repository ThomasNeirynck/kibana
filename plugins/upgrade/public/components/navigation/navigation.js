import React from 'react';

import {
  KuiLocalNav,
  KuiLocalNavRow,
  KuiLocalNavRowSection,
  KuiLocalTab,
  KuiLocalTabs,
} from 'ui_framework/components';

const VIEW_IDS = [
  'HOME',
  'CHECKUP',
  'REINDEX',
  'LOGGING',
];

export function UpgradeSectionNavigation({ currentViewLocation, views }) {
  return (
    <KuiLocalNav>
      <KuiLocalNavRow>
        <KuiLocalNavRowSection>
          <div className="kuiLocalBreadcrumbs">
            <div className="kuiLocalBreadcrumb">
              <a className="kuiLocalBreadcrumb__link" href="#/management">
                Management
              </a>
            </div>
            <h1 id="kui_local_breadcrumb" className="kuiLocalBreadcrumb">
              Upgrade Assistant
            </h1>
          </div>
        </KuiLocalNavRowSection>
      </KuiLocalNavRow>
      <KuiLocalNavRow isSecondary>
        <KuiLocalTabs>
          { VIEW_IDS.map((viewId) => (
            <KuiLocalTab
              href={ views[viewId].absoluteLocation }
              isSelected={ currentViewLocation === views[viewId].location }
              key={ viewId }
            >
              { views[viewId].label }
            </KuiLocalTab>
          ))}
        </KuiLocalTabs>
      </KuiLocalNavRow>
    </KuiLocalNav>
  );
}

UpgradeSectionNavigation.propTypes = {
  currentViewLocation: React.PropTypes.string,
  views: React.PropTypes.object,
};

UpgradeSectionNavigation.defaultProps = {
  currentViewLocation: '',
  views: {},
};
