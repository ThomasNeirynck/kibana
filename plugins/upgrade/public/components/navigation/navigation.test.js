import { render } from 'enzyme';
import React from 'react';

import { UpgradeSectionNavigation } from './navigation';


const VIEWS = {
  HOME: {
    label: 'home label',
    location: 'home',
    absoluteLocation: '/absolute/home/location',
  },
  CHECKUP: {
    label: 'checkup label',
    location: 'checkup',
    absoluteLocation: '/absolute/checkup/location',
  },
  REINDEX: {
    label: 'reindex label',
    location: 'reindex',
    absoluteLocation: '/absolute/reindex/location',
  },
  LOGGING: {
    label: 'logging label',
    location: 'logging',
    absoluteLocation: '/absolute/logging/location',
  },
};

describe('UpgradeSectionNavigation', () => {
  test('renders the views as tabs using their labels and absolute locations', () => {
    const component = (
      <UpgradeSectionNavigation
        views={VIEWS}
      />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders the view as selected if the currentViewLocation matches', () => {
    const component = (
      <UpgradeSectionNavigation
        views={VIEWS}
        currentViewLocation="checkup"
      />
    );

    expect(render(component)).toMatchSnapshot();
  });
});
