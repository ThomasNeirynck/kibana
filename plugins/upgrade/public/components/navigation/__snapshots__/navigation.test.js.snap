// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradeSectionNavigation renders the view as selected if the currentViewLocation matches 1`] = `
<div
  class="kuiLocalNav"
>
  <div
    class="kuiLocalNavRow"
  >
    <div
      class="kuiLocalNavRow__section"
    >
      <div
        class="kuiLocalBreadcrumbs"
      >
        <div
          class="kuiLocalBreadcrumb"
        >
          <a
            class="kuiLocalBreadcrumb__link"
            href="#/management"
          >
            Management
          </a>
        </div>
        <h1
          class="kuiLocalBreadcrumb"
          id="kui_local_breadcrumb"
        >
          Upgrade Assistant
        </h1>
      </div>
    </div>
  </div>
  <div
    class="kuiLocalNavRow kuiLocalNavRow--secondary"
  >
    <div
      class="kuiLocalTabs"
    >
      <a
        class="kuiLocalTab"
        href="/absolute/home/location"
      >
        home label
      </a>
      <a
        class="kuiLocalTab kuiLocalTab-isSelected"
        href="/absolute/checkup/location"
      >
        checkup label
      </a>
      <a
        class="kuiLocalTab"
        href="/absolute/reindex/location"
      >
        reindex label
      </a>
      <a
        class="kuiLocalTab"
        href="/absolute/logging/location"
      >
        logging label
      </a>
    </div>
  </div>
</div>
`;

exports[`UpgradeSectionNavigation renders the views as tabs using their labels and absolute locations 1`] = `
<div
  class="kuiLocalNav"
>
  <div
    class="kuiLocalNavRow"
  >
    <div
      class="kuiLocalNavRow__section"
    >
      <div
        class="kuiLocalBreadcrumbs"
      >
        <div
          class="kuiLocalBreadcrumb"
        >
          <a
            class="kuiLocalBreadcrumb__link"
            href="#/management"
          >
            Management
          </a>
        </div>
        <h1
          class="kuiLocalBreadcrumb"
          id="kui_local_breadcrumb"
        >
          Upgrade Assistant
        </h1>
      </div>
    </div>
  </div>
  <div
    class="kuiLocalNavRow kuiLocalNavRow--secondary"
  >
    <div
      class="kuiLocalTabs"
    >
      <a
        class="kuiLocalTab"
        href="/absolute/home/location"
      >
        home label
      </a>
      <a
        class="kuiLocalTab"
        href="/absolute/checkup/location"
      >
        checkup label
      </a>
      <a
        class="kuiLocalTab"
        href="/absolute/reindex/location"
      >
        reindex label
      </a>
      <a
        class="kuiLocalTab"
        href="/absolute/logging/location"
      >
        logging label
      </a>
    </div>
  </div>
</div>
`;
